Use better command router (dgc)

Muirrum opened the pull request.

Muirrum committed:
Use better command router

Muirrum committed:
Finish admin check

It will now check for the presence of the defined admin role.

samollari commented on middleware/middleware.go:
```diff
@@ -2,44 +2,68 @@ package middleware
+				c.RespondText(fmt.Sprintf("Could not Scan: %s", err))
+				return
+			}
+			err = row.Err()
+			if err != nil {
```
Could you explain how this works and what it does? The intellisense description of `Row.Err()` makes it seem like this will return an error when `Row.Scan` returns an error as well, so that would make the content of the `if` unreachable?

Muirrum replied:
Good catch, this needs to be written to check the user's roles. I'll get on that

samollari replied:
(Again, what's the difference between `err` returned by `Scan` and by `Err`?)

samollari commented on middleware/middleware.go:
```diff
@@ -2,44 +2,68 @@ package middleware
+			if err != nil {
+				guild, _ := c.Session.Guild(c.Event.GuildID)
+				owner := guild.OwnerID
+				if owner == c.Event.Author.ID {
```
This will only allow the owner of the guild to use an admin command? Do we want this to be for anyone with the "Administrator" privilege?

samollari replied:
Seeing the next function, yes it does look like this is the functionality we want.

Muirrum replied:
I thought this was going to be anyone with the role as defined by `minperms`?

samollari replied:
In my original idea, `minperms` was going to determine who can and cannot create certain types of invites. Admin commands were gonna be locked to people with "Create Invites" (I wasn't looking at the doc earlier)

Muirrum replied:
Ohhhh, I misunderstood what you meant in the doc. To be honest, I think that we should just restrict the creation of infinite invites to admins with `minperms`

samollari replied:
Works for me

samollari reviewed (changes requested):
Looks like `AdminOnly` doesn't check for admin privileges? It checks if the user's the guild owner, like `GuildOwnerOnly` does.

Muirrum committed:
Make middleware return after calling command

Avoids case where the middleware will give permission denied message after running command

samollari commented on middleware/middleware.go:
```diff
@@ -36,6 +36,14 @@ func AdminOnly(next dgc.ExecutionHandler) dgc.ExecutionHandler {
+						next(c)
+					}
+				}
+				c.RespondText("You don't have permissions!")
```
Would this respond after `next` finishes? Should it return from the function after calling `next`?

samollari reviewed (changes requested):
Middleware will respond after next handler finished?
